test/integration/stats/data/service_test.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class Stats::Data::ServicesControllerTest < ActionDispatch::IntegrationTest
+  def setup
+    @application = FactoryBot.create :cinstance
+    host! @application.provider_account.admin_domain
+    user = @application.provider_account.admins.first!
+    @token = FactoryBot.create(:access_token, owner: user, scopes: %w[stats], permission: 'rw').value
+  end
+
+  attr_reader :application, :token
+
+  test 'top applications' do
+    params = { access_token: token, since: '2021-08-30 00:00:00', period: :day, metric_name: 'hits' }
```

Comment:
I would suggest another test where `period` is not specified. `usage_params` has a special handling of it.

Reply:
@akostadinov that would be an answer for my question "why should it fail": because it's missing period.